File: src/main/java/com/bank/sentinel/domain/model/Trade.java
package com.bank.sentinel.domain.model;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

public record Trade(
        UUID id,
        String accountId,
        String instrumentId,
        BigDecimal quantity,
        BigDecimal price,
        TradeType type,
        TradeStatus status,
        Instant executedAt
) {}

File: frontend/src/types/trade.ts
export type TradeType = "BUY" | "SELL";

export type TradeStatus = "PENDING" | "PROCESSED" | "FLAGGED" | "REJECTED";

export interface Trade {
  id: string;
  accountId: string;
  instrumentId: string;
  quantity: string;
  price: string;
  type: TradeType;
  status: TradeStatus;
  executedAt: string;
}

export interface AuthTokenResponse {
  token: string;
  expiresIn: number;
}

export interface CreateTradeRequest {
  accountId: string;
  instrumentId: string;
  quantity: string;
  price: string;
  type: TradeType;
}
